raft/src/test/java/org/apache/kafka/raft/KafkaRaftClientTest.java

```diff
@@ -3842,6 +3842,7 @@ public void testMetrics(boolean withKip853Rpc) throws Exception {
         assertNotNull(getMetric(context.metrics, "election-latency-max"));
         assertNotNull(getMetric(context.metrics, "fetch-records-rate"));
         assertNotNull(getMetric(context.metrics, "append-records-rate"));
+        assertNotNull(getMetric(context.metrics, "number-of-voters"));
```

Comment:
We should add a test for the leader specific metric the replica transition from leader to follower to leader. We need to change that after those transition the metrics are reporting the correct value and not the previous value.